ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -634,11 +646,36 @@ object Serializers {
     }
 
     def readTree(): Tree = {
+      val pos = readPosition()
+      readTreeFromTag(input.readByte())(pos)
+    }
+
+    def readOptTree(): Option[Tree] = {
+      if (useHacks068) {
+        readTreeAllowEmpty()
+      } else {
+        import input._
+        if (readBoolean()) {
+          Some(readTree())
+        } else {
+          None
+        }
+      }
+    }
+
+    private def readTreeAllowEmpty(): Option[Tree] = {
+      require(useHacks068)
+      val pos = readPosition()
+      val tag = input.readByte()
+      if (tag == TagEmptyTree) None
+      else Some(readTreeFromTag(tag)(pos))
```

Comment:
Hum, good question. Let's leave it as is. There aren't that many `PropertyDef`s. We'll be able to revisit in 1.0 if we wish.

One thing we could do would be to write/read an `Option[ParamDef]` for the setter argument, and only write/read a `Tree` for the body if the option is `Some(_)`.